File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:

        if n == 1: return 1
        if n == 2: return 2
        
        first, second = 1, 2
        for i in range(3, n + 1):
            temp = first + second
            first = second
            second = temp
        
        return second
        
         

File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        
        if (n == 1) {
            return 1;
        }
        int first = 1;
        int second = 1;
        
        for (int i = 1; i < n; i++) {
            int temp = second;
            second = first + second;
            first = temp;
        }
        
        return second;
    }
};